### DISCUSSION TITLE
invalid refresh token

### DISCUSSION THREAD

**Discussion by iamLazyCode** [dani-garcia/vaultwarden#4304] [General]
hi everyone i keep getting this whats the issue and how to solve...please help 

![image](https://github.com/dani-garcia/vaultwarden/assets/13099434/a20661b1-b88a-4304-87b8-08a19f936db3)

**Comment by BlackDex** [comment]
- Date/Time out of sync on either server or client or both
- The Vaultwarden `rsa_key.pem` file changed, which invalidates all tokens.

Check the logs of the server, that might provide some more details

**Reply by iamLazyCode** [reply]
@BlackDex very thankful for your response sir my knowledge is very limited to linux please if you can guide me more though i never changed anything in rsa key or something

this is the log image

![image](https://github.com/dani-garcia/vaultwarden/assets/13099434/956937af-68e0-45e5-81a0-6222102a4494)

**Reply by BlackDex** [reply]
My guess would be, either a too old server version of Vaultwarden, or a too old Client version of Bitwarden.

**Reply by iamLazyCode** [reply]
@BlackDex 

i did setup vw 10 days back on ubuntu server 22.04

this is the image i used in docker compose `image: vaultwarden/server:latest`
and bitwarden app doesnt shows any updates in play store sir its updated to latest

**Reply by BlackDex** [reply]
Then it might be the reverse proxy which blocks stuff or alters the request in some way.

**Reply by iamLazyCode** [reply]
i did setup using cloud flare and nginx manager

**Reply by BlackDex** [reply]
Might be something in Cloudflare or nginx. Some WAF or ModSecret which is causing issues maybe.

**Reply by iamLazyCode** [reply]
@BlackDex  but sir it worked flawlessly for few days and since yesterday started happening

one thing i see there is a updated pushed yesterday 

is this the right way to update and how can i see currently installed version

```
sudo docker compose down
sudo docker pull vaultwarden/server:latest
sudo docker compose up -d
```

**Reply by BlackDex** [reply]
That is the right way.

You can either check via docker.

```bash
docker exec -it vaultwarden /vaultwarden --version
```

Via the browser:
Admin: `https://vw.domain.tld/admin/diagnostics`
API: `https://vw.domain.tld/api/config`

**Reply by iamLazyCode** [reply]
> That is the right way.
> 
> You can either check via docker.
> 
> ```bash
> docker exec -it vaultwarden /vaultwarden --version
> ```
> 
> Via the browser:
> Admin: `https://vw.domain.tld/admin/diagnostics`
> API: `https://vw.domain.tld/api/config`

Thankyou so much sir and also that issue was sorted i checked everything eas working fine on pc browser just not on android app...i just reinstalled android app and it worked dnt know what went wrong

**Reply by workingwheel** [reply]
Had the same issue on the edge extension. Removing it and installing it again fixed the issue for me.

**Reply by better319** [reply]
invalid refresh token this happened WHEN I STORE THE ACCOUNT FOR ANDROID APP, AND IT'S OK WHEN I STORE THE ACCOUNT OF THE WEBSITE.

**Comment by azazaazur** [comment]
This seems to be fixed by just logging out and relogging to your vaultwarden instance (tested on my iPad). I faced the same issue when I migrated the dataset of vaultwarden’s storage path.
